utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -310,22 +310,37 @@ def get_linux_abi(self, arch):
             'armv7': ('arm', 'gnueabihf')
         }
 
-        # Default is just arch, gnu
-        sysroot_arch, abi = arch_platform_to_abi.get(arch, (arch, 'gnu'))
-        return sysroot_arch, abi
+        abi = 'gnu'
+        vendor = 'unknown'
+
+        try:
+            output = shell.capture(["clang", "--version"])
```

Comment:
This concerns me.  This is dependent on the value of `-D LLVM_DEFALT_TARGET_TRIPLE` passed when building clang.  Worse yet, it simply invokes `clang` without any guarantees on which clang is being invoked.  So, if I were to build a clang and add it to my path even on alpine, this could give me `unknown` as the vendor or any other literal.

Reply:
What would be a more reliable alternative to this invocation then, in your opinion?

Reply:
I would say that you should consider exposing this as a flag that can be explicitly specified, defaulting to processing `/etc/*-release` and `/etc/os-release`, and `lsb_release -b`.  The default would only work under a non-cross-compiling scenario of course, but that applies to most cases for the default.  We could expose this as a completely generic `--default-target` flag perhaps.

Reply:
I've changed it to use `-print-target-triple`, which reliable gives us the actual triple and that's the only output it produces, so we don't need to do any additional parsing.